projects/step-frontend/src/lib/modules/timeseries/components/doughnut-chart/doughnut-chart.component.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+import { DoughnutChartSettings } from './doughnut-chart-settings';
+import Chart from 'chart.js/auto';
+import ChartDataLabels from 'chartjs-plugin-datalabels';
+import { ViewMode } from '../../../execution/shared/view-mode';
+import { StepCommonModule } from '../../../_common/step-common.module';
+
+@Component({
+  selector: 'step-doughnut-chart',
+  templateUrl: './doughnut-chart.component.html',
+  styleUrls: ['./doughnut-chart.component.scss'],
+  imports: [StepCommonModule],
+  standalone: true,
+})
+export class DoughnutChartComponent implements AfterViewInit, OnDestroy, OnChanges {
+  @Input() settings!: DoughnutChartSettings;
+
+  @ViewChild('canvas') private canvas!: ElementRef<HTMLCanvasElement>;
+  private chart: Chart | undefined;
```

Comment:
@lucianbaciu Please use signals API for inputs and viewChild.

chart itself, will be computed signal in this case